Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -329,6 +449,15 @@ public Task<TResult> InvokeWithParameterObjectAsync<TResult>(string targetName,
             return data;
         }
 
+        private async Task OnWorkspaceOpening(object sende, EventArgs e) {
+            if (_workspaceFoldersSupported && IsInitialized && _sentInitialWorkspaceFolders && WorkspaceService.CurrentWorkspace != null) {
+                // Send just this workspace folder. Assumption here is that the language client will be destroyed/recreated on
+                // each workspace open
+                var folder = new WorkspaceFolder { uri = new System.Uri(WorkspaceService.CurrentWorkspace.Location), name = WorkspaceService.CurrentWorkspace.GetName() };
+                await InvokeDidChangeWorkspaceFoldersAsync(new WorkspaceFolder[] { folder }, new WorkspaceFolder[0]);
```

Comment:
probably more of these extra 'await' that could just be returns.  could inverse the if logic to return a completed task but thats more a styling sugguestion

Reply:
Yeah the async means the implicit else on the if statement doesn't need to return a completed task.